gateway-service/src/main/java/org/zowe/apiml/gateway/zaas/ZaasController.java

```diff
@@ -116,4 +118,43 @@ public ResponseEntity<Object> getZoweJwt(@RequestAttribute(AUTH_SOURCE_ATTR) Aut
         }
     }
 
+    @PostMapping(path = "safIdt", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Provides SAF Identity Token for authenticated user.")
+    public ResponseEntity<Object> getSafIdToken(@RequestBody TicketRequest ticketRequest, @RequestAttribute(AUTH_SOURCE_PARSED_ATTR) AuthSource.Parsed authSourceParsed) {
+
+        final String userId = authSourceParsed.getUserId();
+        if (StringUtils.isEmpty(userId)) {
+            return ResponseEntity
+                .status(HttpStatus.UNAUTHORIZED)
+                .build();
+        }
+
+        final String applicationName = ticketRequest.getApplicationName();
+        if (StringUtils.isBlank(applicationName)) {
+            ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.security.ticket.invalidApplicationName").mapToView();
+            return ResponseEntity
+                .status(HttpStatus.BAD_REQUEST)
+                .body(messageView);
+        }
+
+        try {
+            String safIdToken = tokenCreationService.createSafIdTokenWithoutCredentials(userId, applicationName);
+            return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(new ZaasTokenResponse(null, safIdToken));
```

Comment:
Not sure what exactly it returnes when JSON is generated. I would rather use "" ... but as I said maybe it is okay.